### PR TITLE
Dilpreet | Stage | Integrate msg91 api to send mail on customer/vendor page

### PR DESCRIPTION
**Pull request by dilpreetsinghofficial**
https://app.clickup.com/t/85zrpq7xv

### CODE REVIEW

File: apps/web-giddh/src/app/contact/contact.component.html [hunk 1]
```diff
@@ -697,6 +705,13 @@ <h4>{{ account?.gstIn }}</h4>
                                     >
                                         {{ commonLocaleData?.app_send_email }}
                                     </button> -->
+                                     <button
+                                        mat-menu-item
+                                        class="menu-table-item"
+                                        (click)="openCustomEmailDialog(account, activeTab, false)"
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/contact/contact.component.html:711]:
```suggestion
                                        (click)="openCustomEmailDialog(account, activeTab, false)"
                                        aria-label="send email"
```
Optional changes

File: apps/web-giddh/src/app/contact/contact.component.html [hunk 2]
```diff
@@ -1037,6 +1052,13 @@ <h4>{{ account?.gstIn }}</h4>
                                     >
                                         {{ commonLocaleData?.app_record_bill }}
                                     </button>
+                                    <button
+                                        mat-menu-item
+                                        class="menu-table-item"
+                                        (click)="openCustomEmailDialog(account, activeTab, false)"
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/contact/contact.component.html:1060]:
```suggestion
                                        (click)="openCustomEmailDialog(account, activeTab, false)"
                                        aria-label="send email"
```

optional changes

File: apps/web-giddh/src/app/contact/contact.component.ts [hunk 3]
```diff
@@ -239,13 +243,17 @@ export class ContactComponent implements OnInit, OnDestroy {
     public isPlaidSupportedCountry: boolean;
     /** Stores the voucher API version of current company */
     public voucherApiVersion: 1 | 2 = 2;
+    /** Stores the send email bulk request  */
+    public sendBulkEmailRequest: SendBulkEmailTemplateRequest;
+    /** Observable for bulk email success response */
+    public bulkEmailSuccess$ : any = this.componentStore.select(state => state.sendBulkEmailISuccess);
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/contact/contact.component.ts]: correct type and formating

File: apps/web-giddh/src/app/contact/contact.component.ts [hunk 4]
```diff
@@ -239,13 +243,17 @@ export class ContactComponent implements OnInit, OnDestroy {
...
     /** True if consolidated branch */
     public isConsolidatedBranch: boolean;
 
     constructor(public dialog: MatDialog, private store: Store<AppState>, private router: Router, private companyServices: CompanyService, private commonActions: CommonActions, private toaster: ToasterService,
         private contactService: ContactService, private settingsIntegrationActions: SettingsIntegrationActions, private companyActions: CompanyActions, private componentFactoryResolver: ComponentFactoryResolver, private cdRef: ChangeDetectorRef, private generalService: GeneralService, private route: ActivatedRoute, private generalAction: GeneralActions,
         private breakPointObservar: BreakpointObserver, private modalService: BsModalService, private settingsProfileActions: SettingsProfileActions,
-        private settingsBranchAction: SettingsBranchActions, public currencyPipe: GiddhCurrencyPipe, private lightbox: Lightbox, private renderer: Renderer2) {
+        private settingsBranchAction: SettingsBranchActions, public currencyPipe: GiddhCurrencyPipe, private lightbox: Lightbox, private renderer: Renderer2, private componentStore: ContactComponentStore,) {
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/contact/contact.component.ts]:
```suggestion
        private settingsBranchAction: SettingsBranchActions, public currencyPipe: GiddhCurrencyPipe, private lightbox: Lightbox, private renderer: Renderer2, private componentStore: ContactComponentStore) {
```

File: apps/web-giddh/src/app/contact/contact.component.ts [hunk 5]
```diff
@@ -291,7 +299,18 @@ export class ContactComponent implements OnInit, OnDestroy {
...
-
+        this.bulkEmailSuccess$.pipe(
+            takeUntil(this.destroyed$)
+        ).subscribe(success => {
+            if (success) {
+                this.selectedCheckedContacts = [];
+                this.selectedAccountsList = [];
+                this.allSelectionModel = false;
+                this.checkboxInfo = {
+                    selectedPage: 1,
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/contact/contact.component.ts]:
```suggestion
                    selectedPage: 1
```

File: apps/web-giddh/src/app/contact/contact.component.ts [hunk 6]
```diff
@@ -1717,4 +1736,49 @@ export class ContactComponent implements OnInit, OnDestroy {
...
+    }
+
+    /**
+   * Open custom email dialog
+   *
+   * @param {any} account
+   * @memberof ContactComponent
+   */
+    public openCustomEmailDialog(account: any, activeTab: string, sendBulk: boolean): void {
+        const dialogRef =  this.dialog.open(TemplateFroalaComponent, {
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/contact/contact.component.ts]:
```suggestion
        const dialogRef = this.dialog.open(TemplateFroalaComponent, {
```

File: apps/web-giddh/src/app/contact/utility/contact.store.ts [hunk 7]
```diff
@@ -0,0 +1,74 @@
+
+import { Injectable, OnDestroy } from "@angular/core";
+import { ComponentStore, tapResponse } from "@ngrx/component-store";
+import { Observable, switchMap, catchError, EMPTY } from "rxjs";
+import { BaseResponse } from "../../models/api-models/BaseResponse";
+import { ToasterService } from "../../services/toaster.service";
+import { ContactService } from "../../services/contact.service";
+
+export interface ContactState {
+    sendBulkEmailISuccess: boolean;
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/contact/utility/contact.store.ts]:
```suggestion
    sendBulkEmailIsSuccess: boolean;
```

File: apps/web-giddh/src/app/contact/utility/contact.store.ts [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+     */
+    readonly sendBulkEmailTemplate = this.effect((data: Observable<any>) => {
+        return data.pipe(
+            switchMap((req) => {
+                this.patchState({ sendBulkEmailISuccess: false });
+                return this.contactService.sendBulkEmailTemplate(req).pipe(
+                    tapResponse(
+                        (res: BaseResponse<any, any>) => {
+                            if (res?.status === 'success') {
+                                this.toasterService.showSnackBar('success', res?.body);
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/contact/utility/contact.store.ts]:
```suggestion
                                res?.body && this.toasterService.showSnackBar('success', res.body);
```

File: apps/web-giddh/src/app/contact/utility/contact.store.ts [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+                                return this.patchState({
+                                    sendBulkEmailISuccess: true
+                                });
+                            } else {
+                                if (res.message) {
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/contact/utility/contact.store.ts]:
```suggestion
                                if (res?.message) {
```

File: apps/web-giddh/src/app/models/api-models/Contact.ts [hunk 10]
```diff
@@ -203,3 +203,14 @@ export class BulkUpdateInvoiceCustomfields {
...
+ * Send Bulk Email Template request model
+ *
+ * @export
+ * @interface SendEmailTemplateRequest
+ */
+export interface SendBulkEmailTemplateRequest {
+    customerVendorUniqueNames: string[];
+    templateOf: string;
+}
+
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/models/api-models/Contact.ts]:
```suggestion
```

File: apps/web-giddh/src/app/services/apiurls/contact.api.ts [hunk 11]
```diff
@@ -1,4 +1,5 @@
 export const CONTACT_API = {
     ADD_COMMENT: 'v2/company/:companyUniqueName/accounts/:accountUniqueName/report-comment',
-    GOCARDLESS_BANKTRANSACTIONS_REFRESH: 'company/:companyUniqueName/gocardless/refresh'
+    SEND_EMAIL_TEMPLATE: 'v2/company/:companyUniqueName/accounts/send-mail',
+    GOCARDLESS_BANK_TRANSACTIONS_REFRESH: 'company/:companyUniqueName/gocardless/refresh'
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/services/apiurls/contact.api.ts:4]: move in line 3 as previous

File: apps/web-giddh/src/app/services/contact.service.ts [hunk 12]
```diff
@@ -127,14 +129,31 @@ export class ContactService {
...
     }
 
+    /**
+     * Send bulk email template to specified customers or vendors
+     * @param model Request payload containing customer/vendor unique names and template type
+     * @returns Observable<BaseResponse<any, string>> API response
+     */
+    public sendBulkEmailTemplate(model: SendBulkEmailTemplateRequest): Observable<BaseResponse<any, string>> {
+        this.companyUniqueName = this.generalService.companyUniqueName;
+        return this.http.post(this.config.apiUrl + CONTACT_API.SEND_EMAIL_TEMPLATE?.replace(':companyUniqueName', encodeURIComponent(this.companyUniqueName)), model).pipe(
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/services/contact.service.ts]:
```suggestion
        return this.http.post(this.config.apiUrl + CONTACT_API.SEND_EMAIL_TEMPLATE?.replace(':companyUniqueName', encodeURIComponent(this.generalService.companyUniqueName)), model).pipe(
```

File: apps/web-giddh/src/app/services/contact.service.ts [hunk 13]
```diff
@@ -127,14 +129,31 @@ export class ContactService {
         }
     }
 
+    /**
+     * Send bulk email template to specified customers or vendors
+     * @param model Request payload containing customer/vendor unique names and template type
+     * @returns Observable<BaseResponse<any, string>> API response
+     */
+    public sendBulkEmailTemplate(model: SendBulkEmailTemplateRequest): Observable<BaseResponse<any, string>> {
+        this.companyUniqueName = this.generalService.companyUniqueName;
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/services/contact.service.ts]:
```suggestion
```

File: apps/web-giddh/src/app/shared/template-froala/template-froala.component.ts [hunk 14]
```diff
@@ -162,7 +168,8 @@ export class TemplateFroalaComponent implements OnInit {
 
         this.emailContentSuggestions$.pipe(takeUntil(this.destroyed$)).subscribe(response => {
             if (response) {
-                const tributeSuggestions = response?.voucherSuggestions?.map(item => ({
+                const emailSuggestions = this.inputData.activeTab ? response?.customerVendorSuggestions : response?.voucherSuggestions;
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/shared/template-froala/template-froala.component.ts]:
```suggestion
                const emailSuggestions = this.inputData?.activeTab ? response?.customerVendorSuggestions : response.voucherSuggestions;
```

File: apps/web-giddh/src/app/theme/form-fields/select-multiple-fields/select-multiple-fields.component.ts [hunk 15]
```diff
@@ -165,6 +165,12 @@ export class SelectMultipleFieldsComponent implements OnInit, OnDestroy, OnChang
     public removeOption(index: number): void {
         if (index >= 0) {
             this.chipList.splice(index, 1);
+            // Close the autocomplete dropdown if it's open
+            setTimeout(() => {
+                if (this.trigger && this.trigger.panelOpen) {
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/theme/form-fields/select-multiple-fields/select-multiple-fields.component.ts:170]: setTimeout k time agr this.trigger milta tha to set timeout ko apn if block andr dal skte h

**Review by dvCodeWorld — CHANGES_REQUESTED** [review]

File: apps/web-giddh/src/app/theme/form-fields/select-multiple-fields/select-multiple-fields.component.ts (hunk 15 above)
Comment (dilpreetsinghofficial) [reply apps/web-giddh/src/app/theme/form-fields/select-multiple-fields/select-multiple-fields.component.ts:170]: nh milta h

File: apps/web-giddh/src/app/services/apiurls/contact.api.ts (hunk 11 above)
Comment (dilpreetsinghofficial) [reply apps/web-giddh/src/app/services/apiurls/contact.api.ts:4]: dont understand

File: apps/web-giddh/src/app/contact/contact.component.ts [hunk 16]
```diff
@@ -246,14 +246,14 @@ export class ContactComponent implements OnInit, OnDestroy {
     /** Stores the send email bulk request  */
     public sendBulkEmailRequest: SendBulkEmailTemplateRequest;
     /** Observable for bulk email success response */
-    public bulkEmailSuccess$ : any = this.componentStore.select(state => state.sendBulkEmailISuccess);
+    public bulkEmailSuccess$: Observable<any> = this.componentStore.select(state => state.sendBulkEmailIsSuccess);
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/contact/contact.component.ts]: It should be boolean instead of any

**Review by dvCodeWorld — APPROVED** [review]

**Pull request merged**